lua/cmp/autocmds.lua

```diff
@@ -0,0 +1,130 @@
+local M = {}
+
+local v = vim.version()
+local has_native_autocmd = v.major >= 0 and v.minor >= 7
+local emit = require('cmp.utils.autocmd').emit
+
+local function nvim_auto()
+  local ___cmp___ = vim.api.nvim_create_augroup('___cmp___', { clear = true })
+  vim.api.nvim_create_autocmd('InsertEnter', {
+    group = ___cmp___,
+    callback = function()
+      emit('InsertEnter')
+    end,
+  })
+
+  vim.api.nvim_create_autocmd('InsertLeave', {
+    group = ___cmp___,
+    callback = function()
+      emit('InsertLeave')
+    end,
+  })
+
+  vim.api.nvim_create_autocmd({
+    'TextChangedI',
+    'TextChangedP',
+  }, {
+
+    group = ___cmp___,
+    callback = function()
+      emit('TextChanged')
+    end,
+  })
+
+  vim.api.nvim_create_autocmd('CursorMovedI', {
+    group = ___cmp___,
+    callback = function()
+      emit('CursorMoved')
+    end,
+  })
+
+  vim.api.nvim_create_autocmd('CompleteChanged', {
+    group = ___cmp___,
+    callback = function()
+      emit('CompleteChanged')
+    end,
+  })
+
+  vim.api.nvim_create_autocmd('CompleteDone', {
+    group = ___cmp___,
+    callback = function()
+      emit('CompleteDone')
+    end,
+  })
+
+  vim.api.nvim_create_autocmd('ColorScheme', {
+    group = ___cmp___,
+    callback = _G.cmp.plugin.colorscheme,
+  })
+
+  -- for entering cmdwin with `<C-f>`
+  vim.api.nvim_create_autocmd('CmdlineEnter', {
+    group = ___cmp___,
+    callback = _G.cmp.plugin.cmdline.enter,
+  })
+
+  vim.api.nvim_create_autocmd('CmdwinEnter', {
+    group = ___cmp___,
+    callback = _G.cmp.plugin.cmdline.leave,
+  })
+end
+
+local function legacy_auto()
```

Comment:
Sorry but the nvim0.u is released so we don't need fallback to old implementation.
Please use only vim.api.*